exchanges/stream/stream_match.go

```diff
@@ -9,21 +9,22 @@ var errSignatureCollision = errors.New("signature collision")
 
 // NewMatch returns a new Match
 func NewMatch() *Match {
-	return &Match{m: make(map[any]chan<- []byte)}
+	return &Match{m: make(map[any]Incoming)}
 }
 
 // Match is a distributed subtype that handles the matching of requests and
 // responses in a timely manner, reducing the need to differentiate between
 // connections. Stream systems fan in all incoming payloads to one routine for
 // processing.
 type Match struct {
-	m  map[any]chan<- []byte
+	m  map[any]Incoming
 	mu sync.Mutex
 }
 
-// Incoming matches with request, disregarding the returned payload
-func (m *Match) Incoming(signature any) bool {
-	return m.IncomingWithData(signature, nil)
+// Incoming is a sub-type that handles incoming data
+type Incoming struct {
+	count          int // Number of responses expected
+	waitingRoutine chan<- []byte
```

Comment:
This isn't a waitingRoutine; We actually don't know anything at all about the consumers/readers, which is good.

So I think we should just call it `c`, which has a lot of precedent in go already.